server/game/core/attack/AttackFlow.ts

```diff
@@ -21,13 +18,22 @@ export class AttackFlow extends BaseStepWithPipeline {
         this.pipeline.initialise([
             new SimpleStep(this.game, () => this.setCurrentAttack(), 'setCurrentAttack'),
             new SimpleStep(this.game, () => this.declareAttack(), 'declareAttack'),
-            new SimpleStep(this.game, () => this.dealDamage(), 'dealDamage'),
+            new SimpleStep(this.game, () => this.openDealDamageWindow(), 'openDealDamageWindow'),
             new SimpleStep(this.game, () => this.completeAttack(), 'completeAttack'),
             new SimpleStep(this.game, () => this.cleanUpAttack(), 'cleanUpAttack'),
             new SimpleStep(this.game, () => this.game.resolveGameState(true), 'resolveGameState')
         ]);
     }
 
+    private openDealDamageWindow(): void {
```

Comment:
nit: do you mind moving this below `declareAttack` so it lines up with the order the pipeline steps are declared in

Reply:
moved below declareAttack to make it look more clean.